### DISCUSSION TITLE
Test if input is a number.

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#340] [General]
I have a text field that where I want users to only be able to input numbers and not letters.  Any suggestions on what I should think about or Google on how to perform such testing?  What might be the best approach?

**Comment by rabbibotton** [comment]
;;;;;;;;;;;;;
;; pattern ;;
;;;;;;;;;;;;;

(defgeneric pattern (clog-form-element)
  (:documentation "Get/Setf form element regular expression pattern.
   (see JavaScript RegExp object for specifics)
   Form validation pattern. validate-on-submit fields with input
   will validate against their pattern if set on submit.
   In cases where a specific input type is not supported like
   (date, week, etc.) Pattern can be set to ensure the expected results.
   This works since Input type will fall back to a text input."))

(defmethod pattern ((obj clog-form-element))
  (property obj "pattern"))

(defgeneric (setf pattern) (value clog-form-element)
  (:documentation "Set pattern VALUE for CLOG-FORM-ELEMENT"))

(defmethod (setf pattern) (value (obj clog-form-element))
  (setf (property obj "pattern") value))


example -  "[A-Za-z]{3}" only letters max 3

**Comment by aykaramba** [comment]
Ohhh, I see.  Okay great, thank you kindly.
